scipy/stats/_continuous_distns.py

```diff
@@ -6973,6 +6973,171 @@ def _stats(self, a):
+            # value of the shape.
+            return -len(data) / np.sum(np.log((data - loc)/scale))
+
+        def get_scale(data, loc):
+            # analytical solution for `scale` based on the location.
+            # It can be used no matter the assumption of the value of the
+            # shape.
+            return data.max() - loc
+
+        # 1) The location and scale are both fixed. Analytically determine the
+        # shape.
+        if fscale is not None and floc is not None:
+            return get_shape(data, floc, fscale), floc, fscale
+
+        # 2) The scale is fixed. The scale is one of two possible analytical
+        # solutions. Choose the one with better log-likelihood.
+        if fscale is not None:
+            args = [data, (self._fitstart(data),)]
+            func = self._reduce_func(args, {})[1]
+
+            # using `data.min()` as the optimal location
+            loc_lt1 = np.nextafter(data.min(), -np.inf)
+            shape_lt1 = fshape or get_shape(data, loc_lt1, fscale)
+            ll_lt1 = func((shape_lt1, loc_lt1, fscale), data)
+
+            # using `data.max() - scale` as the optimal location
+            loc_gt1 = np.nextafter(data.max() - fscale, np.inf)
+            shape_gt1 = fshape or get_shape(data, loc_gt1, fscale)
+            ll_gt1 = func((shape_gt1, loc_gt1, fscale), data)
+
+            if ll_lt1 < ll_gt1:
+                return shape_lt1, loc_lt1, fscale
+            else:
+                return shape_gt1, loc_gt1, fscale
+
+        # 3) The location is fixed. Return the analytical scale and the
+        # analytical (or fixed) shape.
+        if floc is not None:
+            scale = get_scale(data, floc)
+            shape = fshape or get_shape(data, floc, scale)
+            return shape, floc, scale
+
+        # 4) location, scale, and shape are all free. Attempt to fit under the
+        # assumption that `shape <= 1` analytically.
+        loc = np.nextafter(data.min(), -np.inf)
+        scale = np.nextafter(get_scale(data, loc), np.inf)
+        shape = fshape or get_shape(data, loc, scale)
+
+        # If the `shape <= 1`, then fitting has succeeded with the assumption
+        # `shape <= 1`.
+        if 0 < shape <= 1:
+            return shape, loc, scale
+
+        # 5) Attempt to fit under the assumption that `shape > 1`. The support
+        # of the distribution is `(x - loc)/scale > 0`. If all parameters are
+        # free, or only the shape is fixed, the method of Lagrange multipliers
+        # turns this constraint into an equation that can be solved
+        # numerically.
+
+        def dL_dLocation(data, shape, scale):
+            # The partial derivative of the log-likelihood function w.r.t.
+            # the location.
+            return - data.shape[0] * shape / scale
+
+        def dL_dScale(data, shape, loc):
+            # The partial derivative of the log-likelihood function w.r.t.
+            # the scale.
+            mask = data != loc
```

Comment:
The correct solution, which we've implemented, is to ensure that `loc` does not equal any elements of the `data`, not omitting the elements of `data` that equal `loc`. It would have been inaccurate to omit values from the data.